### PR TITLE
fix: Delete all port_id references #4684

### LINKED ISSUE #4684: delete_port does not delete all entries referencing a port
DO NOT DELETE THIS INFORMATION.

> Please read this information carefully.

- [x] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
      Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
- [ ] Please include all of the information between the `====================================` section of `./validate.php`.
- [ ] If you would like us to add a new device then please provide the information asked for [here](http://docs.librenms.org/Support/FAQ/#faq20)
- [ ] Please provide as much detail as possible.

I just noticed that delete_port (called from delete_device) does not delete everything related to a port.

It is deleting data from about 8 tables, but there seem to be a lot more that reference port_id in some way or another.  I would just add them all to the list, but I'm not sure if that is the desired behavior (also, links and ports_stack will need special handling).

```
MariaDB [(none)]> SELECT DISTINCT TABLE_NAME
     FROM INFORMATION_SCHEMA.COLUMNS
     WHERE COLUMN_NAME LIKE '%port_id%'
     AND TABLE_SCHEMA='librenms';
+--------------------+
| TABLE_NAME         |
+--------------------+
| bill_port_counters |
| bill_ports         |
| ipv4_addresses     |
| ipv4_mac           |
| ipv6_addresses     |
| juniAtmVp          |
| links              |
| mac_accounting     |
| ospf_nbrs          |
| ospf_ports         |
| ports              |
| ports_adsl         |
| ports_perms        |
| ports_stack        |
| ports_statistics   |
| ports_stp          |
| ports_vlans        |
| pseudowires        |
+--------------------+
18 rows in set (0.00 sec)

MariaDB [(none)]> 
```

### PR DESCRIPTION
**Pull request by Rosiak**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

fix #4684

### CODE REVIEW

File: includes/common.php [hunk 1]
```diff
@@ -140,15 +140,14 @@ function delete_port($int_id)
 {
     $interface = dbFetchRow("SELECT * FROM `ports` AS P, `devices` AS D WHERE P.port_id = ? AND D.device_id = P.device_id", array($int_id));
 
-    $interface_tables = array('adjacencies', 'ipaddr', 'ip6adjacencies', 'ip6addr', 'mac_accounting', 'bill_ports', 'pseudowires', 'ports');
+    $interface_tables = array('bill_ports', 'bill_port_counters', 'ipv4_addresses', 'ipv4_mac', 'ipv6_addresses', 'juniAtmVp', 'mac_accounting', 'ospf_nbrs', 'ospf_ports', 'ports', 'ports_adsl', 'ports_perms', 'ports_statistics', 'ports_stp', 'ports_vlans', 'port_in_measurements', 'port_out_measurements', 'pseudowires');
```
Comment (laf) [inline includes/common.php]: Does this mean then historical billing data is lost? I'm not sure that's a good idea.

Comment (Rosiak) [inline includes/common.php]: How so? Until now we have not cleaned up the bill_port_counters table.

Comment (laf) [inline includes/common.php]: Yeah that's my point. Billing data should. Be left until bills are removed

Comment (Rosiak) [inline includes/common.php]: I guess this would need to be adopted into the remove bills functions(Or however it's done), instead of being here then :)

Comment (laf) [inline includes/common.php]: Yea definitely. If you remove it this PR is probably good to go.

File: includes/common.php [hunk 2]
```diff
@@ -140,15 +140,14 @@ function delete_port($int_id)
 {
     $interface = dbFetchRow("SELECT * FROM `ports` AS P, `devices` AS D WHERE P.port_id = ? AND D.device_id = P.device_id", array($int_id));
 
-    $interface_tables = array('adjacencies', 'ipaddr', 'ip6adjacencies', 'ip6addr', 'mac_accounting', 'bill_ports', 'pseudowires', 'ports');
+    $interface_tables = array('ipv4_addresses', 'ipv4_mac', 'ipv6_addresses', 'juniAtmVp', 'mac_accounting', 'ospf_nbrs', 'ospf_ports', 'ports', 'ports_adsl', 'ports_perms', 'ports_statistics', 'ports_stp', 'ports_vlans', 'port_in_measurements', 'port_out_measurements', 'pseudowires');
```
Comment (laf) [inline includes/common.php]: Just noticed port_in_measurements and port_out_measurements don't exist, they were dropped in 107.sql

Comment (Rosiak) [inline includes/common.php]: Absolutely right, guess my dev db schema needs a refresh.

**Pull request merged**